### PR TITLE
Cypress json schema 1105

### PR DESCRIPTION
**Pull request by bahmutov**
Adding schema file for `cypress.json` allows an editor that supports it to show intellisense for each option. Works really nice. Need to update tests

### CODE REVIEW

File: packages/server/lib/util/settings.coffee [hunk 1]
```diff
@@ -13,6 +13,11 @@ log      = require("../log")
 
 fs = Promise.promisifyAll(fs)
 
+# TODO move default cypress.json out from here into scaffolding for example
+defaultCypressConfig = {
+  "$schema": "./cypress/cypress-schema.json"
```
Comment (brian-mann) [inline packages/server/lib/util/settings.coffee]: Why is this necessary? I am **very** against changing the default `cypress.json` to include a property. I didn't know what a `cypress-schema.json` was before today and I had no idea how it could or couldn't be useful.

Seeing this seeded as the default property would be confusing for the majority of our users.

Comment (bahmutov) [reply packages/server/lib/util/settings.coffee]: maybe it would be confusing. But maybe 99% of users do not look at `cypress.json` file at all. And the ones that try to edit this file would benefit greatly from intellisense that tells what options ARE available and what they do. 

Then other long term solution would be
- to put schema on http://schemastore.org/json/ but then it would not live close to the `cypress.json` and I don't know what to do about versioning
- editor-specific file, but then we would need to put these files out, no one is going to copy / use them

**Comment by bahmutov** [comment]
on a personal note, once I discovered JSON intellisense in VSCode I love it. I edit both `tsconfig.json` and `package.json` using suggestions without opening any docs - and it is super helpful.

**Comment by bahmutov** [comment]
I could make a pull request with this file to https://github.com/schemastore/schemastore/ instead though.

**Comment by bahmutov** [comment]
Will do pull request to schema store instead
